### PR TITLE
Add helm packaging to create_source_release.sh script

### PR DESCRIPTION
**Pull request by gyfora**
This is the logic I have used to create the first RC, formalizing in a script here

### CODE REVIEW

File: tools/releasing/create_source_release.sh [hunk 1]
```diff
@@ -77,10 +77,29 @@ rsync -a \
...
+# Package sources
 tar czf ${RELEASE_DIR}/flink-kubernetes-operator-${RELEASE_VERSION}-src.tgz flink-kubernetes-operator-${RELEASE_VERSION}
 gpg --armor --detach-sig ${RELEASE_DIR}/flink-kubernetes-operator-${RELEASE_VERSION}-src.tgz
+
+# Package helm chart
+commit_hash=$(git log -1 --pretty=format:%h)
+
+# TODO: We might want to be more specific here later on what to replace
+perl -pi -e "s#^  repository: .*#  repository: ghcr.io/apache/flink-operator#" helm/flink-operator/values.yaml
+perl -pi -e "s#^  tag: .*#  tag: ${commit_hash}#" helm/flink-operator/values.yaml
```
Comment (wangyang0918) [inline tools/releasing/create_source_release.sh:89]: We are using commit hash as the image tag, not the `${RELEASE_VERSION}`. Right?

Comment (gyfora) [reply tools/releasing/create_source_release.sh:89]: exactly, we publish the image build with the short commit hash

**Comment by gyfora** [comment]
@wangyang0918 I also added the helm repo index generation logic so users can use the release as a helm repo as you linked for airflow

File: tools/releasing/create_source_release.sh (hunk 1 above)
Comment (mbalassi) [reply tools/releasing/create_source_release.sh:89]: We can certainly improve this once we get some more experience with this, but currently this is the approach we can use for this preview release candidate.

**Review by mbalassi — APPROVED** [review]
LGTM

**Review by wangyang0918 — APPROVED** [review]
+1

File: tools/releasing/create_source_release.sh (hunk 1 above)
Comment (wangyang0918) [reply tools/releasing/create_source_release.sh]: It seems that the image tag in [rc2](https://dist.apache.org/repos/dist/dev/flink/flink-kubernetes-operator-0.1.0-rc2/) is `260df17`. But I could not find this commit in release-0.1. Is this the expected behavior?

Comment (gyfora) [reply tools/releasing/create_source_release.sh]: The commit recorded in the RC is the commit of the release tag: https://github.com/apache/flink-kubernetes-operator/tree/release-0.1.0-rc2

Comment (wangyang0918) [reply tools/releasing/create_source_release.sh]: You are correct.

**Pull request merged**